src/mainloop.cc

```diff
@@ -0,0 +1,64 @@
+#include "mainloop.h"
+
+#include <uv.h>
+#include <glib.h>
+
+using namespace v8;
+
+struct ThreadData {
+  uv_thread_t thread_handle;
+  uv_mutex_t mutex_handle;
+  uv_async_t async_handle;
+  GPollFD *fds;
+  gint nfds;
+};
+
+static void DispatchGLibMainloop (uv_async_t *async_handle) {
```

Comment:
I don't understand too well what's going on here, maybe an overview for dummies?